lib/ansible/modules/network/fortios/fortios_config.py

```diff
@@ -0,0 +1,211 @@
+        changes are made.  The backup file is written to the C(backup)
+        folder in the playbook root directory.  If the directory does not
+        exist, it is created.
+    default: no
+    choices: ['yes', 'no']
+notes:
+  - This module requires pyFG python library
+"""
+
+EXAMPLES = """
+- name: Backup current config
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    backup: yes
+
+- name: Backup only address objects
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    backup: yes
+    filter: "firewall address"
+
+- name: Update configuration from file
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    src: new_configuration.conf
+
+"""
+
+RETURN = """
+running_config:
+  description: full config string
+  returned: always
+  type: string
+change_string:
+  description: The commands really executed by the module
+  returned: only if config changed
+  type: string
+"""
+
+import os
+import time
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+
+#check for pyFG lib
+try:
+    from pyFG import FortiOS, FortiConfig
+    from pyFG.fortios import logger
+    from pyFG.exceptions import CommandExecutionException, FailedCommit, ForcedCommit
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+# some blocks don't support update, so remove them
+NOT_UPDATABLE_CONFIG_OBJECTS=[
+    "vpn certificate local",
+]
+
+def main():
+    argument_spec = dict(
+        host      = dict(required=True ),
+        username  = dict(required=True ),
+        password  = dict(required=True, type='str', no_log=True ),
+        timeout   = dict(type='int', default=60),
+        vdom      = dict(type='str', default=None ),
+        src       = dict(type='str', default=None),
+        filter    = dict(type='str', default=""),
+        backup    =dict(type='bool', default=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    result = dict(changed=False)
+
+    # fail if pyFG not present
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    #connect
+    try:
+        f.open()
+    except:
+        module.fail_json(msg='Error connecting device')
+
+    #get  config
+    try:
+        f.load_config(path=module.params['filter'])
+        result['running_config'] = f.running_config.to_text()
+
+    except:
+        module.fail_json(msg='Error reading running config')
+
+    #backup config
+    if module.params['backup']:
+        backup_path = 'backup'
+        if not os.path.exists(backup_path):
```

Comment:
regarding backup, the way this is implemented will backup the file to the location where the module is executed. When we exeucte this as local_action or  `ansible_connection=local`, it will backup the file to the control machine, however we run this module with `ansible_connection=ssh` it will be backuped to the remote machine. 

So this should be solved as "action_plugin" (identical to ios_config) which will ensure, the backup is going to the control machine.